### ISSUE TITLE
Error case of a known signal asked but for an object type that doesn't match

### ISSUE THREAD

**Issue by marcgarba** [rsmp-nordic/rsmp_core#163]
Hello,

If we have the case of a request for a known signal code id , but for an object type that doesn't match, this error handling case is not listed in the corresponding chapter of RSMP Core.
Currently the simulators seems to respond with a nack, so this case is considered as a SXL mismatch error.

Example with a detector object (and a controller signal): 
```
2024-04-09 17:33:20.911	Rx <-	{"mType":"rSMsg",
		"type":"StatusRequest",
		"mId":"61e9e189-7728-4568-86dc-97f907652f33",
		"ntsOId":"AA+BBCCC=DDDEEFFF",
		"xNId":"",
		"cId":"AA+BBCCC=DDDDLAA1",
		"sS":[{"sCI":"S0002",
		"n":"detectorlogicstatus"}]}

2024-04-09 17:33:20.911	Tx ->	{"mType":"rSMsg",
		"type":"MessageNotAck",
		"oMId":"61e9e189-7728-4568-86dc-97f907652f33",
		"rea":"Got Status, failed to find sCI (NTSObjectId: AA+BBCCC=DDDEEFFF,
		 ComponentId: AA+BBCCC=DDDDLAA1, StatusCodeId: S0002, Name: detectorlo
		gicstatus)"}
```

The rea error "signal code not found" is perhaps the simulator only search in the signals for the corresponding object type...
(we could also imagine a possible different error : "Mismatch object type for sCI asked")

Thanks.

**Comment by otterdahl** [comment]
> If we have the case of a request for a known signal code id , but for an object type that doesn't match, this error handling case is not listed in the corresponding chapter of RSMP Core. Currently the simulators seems to respond with a nack, so this case is considered as a SXL mismatch error.

I think this case qualifies as a [SXL mismatch](https://rsmp-nordic.github.io/rsmp_core/3.2.2/applicability/error_handling.html#sxl-mismatch) where the status code id is unknown for the specified object type. 

> The rea error "signal code not found" is perhaps the simulator only search in the signals for the corresponding object type... (we could also imagine a possible different error : "Mismatch object type for sCI asked")

I think the simulator actually allows for a separate set of signal code id's for each object type. So I think it is possible to use a separate S0001 for both a signal group as well as a detector logic at same time. But I think we've decided to define status code id's to be unique for the SXL to avoid confusion. But this may explain why the simulators responds with a MessageNotAck with the given "reason" text as you noticed.

I suppose we could update the simulators to send a different error message if it would clarify things better.

**Comment by marcgarba** [comment]
Perhaps this case could at least be added in the "4.5.2 SXL mismatch" :
- known code id (alarm, status, command) but not corresponding to the object type.
To think about it...

**Comment by otterdahl** [comment]
How about changing this text:

* unknown alarm code id (``aCId``)
* unknown status code id (``sCI``)
* unknown command code id (``cCI``)
* unknown name (``n``) in arguments or return values

to this:

* unknown alarm code id (``aCId``) for the corresponding object type
* unknown status code id (``sCI``) for the corresponding object type
* unknown command code id (``cCI``) for the corresponding object type
* unknown name (``n``) in arguments or return values

or perhaps making is simpler:

* unknown alarm/status/command code id (``aCId``, ``sCI``, ``cCI``) for the corresponding object type
* unknown name (``n``) in arguments or return values

**Comment by marcgarba** [comment]
The simpler is perhaps the better !

**Comment by otterdahl** [comment]
I've added a PR: https://github.com/rsmp-nordic/rsmp_core/pull/164

**Comment by otterdahl** [comment]
The PR has been merged to the 3.2.2 draft. You can view the updated draft here: https://rsmp-nordic.github.io/rsmp_core/3.2.2/applicability/error_handling.html#sxl-mismatch
